metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -0,0 +1,188 @@
+                    get_attribute_definitions(self.config.dynamodbstreams_client, stream_arn)
+                )
+            else:
+                logger.error(f"Table {table_name} streaming is NOT enabled.")
+                self.report.report_table_streaming_disabled(table_name)
+            dataset_urn = make_dataset_urn(DEFAULT_PLATFORM, table_name, self.config.env)
+            schema_fields = self.get_schema_fields(attribute_definitions)
+            schema_metadata = SchemaMetadata(
+                schemaName=f"{table_name}-schema",
+                platform=make_data_platform_urn(DEFAULT_PLATFORM),
+                version=0,
+                hash="",
+                platformSchema=KeyValueSchema(keySchema="", valueSchema=""),
+                fields=schema_fields,
+                primaryKeys=keys
+            )
+
+            mcp = MetadataChangeProposalWrapper(
+                entityType="dataset",
+                entityUrn=dataset_urn,
+                changeType=ChangeTypeClass.UPSERT,
+                aspectName="schemaMetadata",
+                aspect=schema_metadata
+            )
+            wu = MetadataWorkUnit(
+                id=dataset_urn,
+                mcp=mcp
+            )
+            yield wu
+
+
+    def get_schema_fields(self, attribute_definitions: Dict):
+        schema_fields = []
+        for (k, v) in attribute_definitions.items():
+            type_class = _attribute_type_mapping.get(v)
+            schemaField = SchemaField(
+                fieldPath=k,
+                type=SchemaFieldDataType(type=type_class()),
+                nativeDataType=v
+            )
+            schema_fields.append(schemaField)
+        return schema_fields
+
+
+    def get_report(self):
+        return self.report
+
+
+    def close(self):
+        pass
+
+
+def get_all_tables(dynamodb_client) -> List[str]:
+    tables = []
+    paginator = dynamodb_client.get_paginator("list_tables")
+    paginator_response = paginator.paginate()
+    for page in paginator_response:
+        tables += page["TableNames"]
+    return tables
+
+
+def describe_table(dynamodb_client, table_name: str) -> Dict:
+    response = dynamodb_client.describe_table(TableName=table_name)
+    table = response["Table"]
+    return table
+
+
+def get_attribute_definitions(dynamodbstreams_client, arn: str) -> Dict:
+    attribute_definitions = OrderedDict()
+    description_response = dynamodbstreams_client.describe_stream(StreamArn=arn)
+    description = description_response.get("StreamDescription", {})
+    shards = description.get("Shards", [])
+    for shard in shards:
+        shard_iterator_response = dynamodbstreams_client.get_shard_iterator(
+            StreamArn=arn,
+            ShardId=shard["ShardId"],
+            ShardIteratorType="TRIM_HORIZON"
+        )
+        shard_iterator = shard_iterator_response["ShardIterator"]
+        records_response = dynamodbstreams_client.get_records(ShardIterator=shard_iterator)
```

Comment:
Documentation says:
`GetRecords can retrieve a maximum of 1 MB of data or 1000 stream records, whichever comes first.`

So we won't get everything, and it is would be pretty memory intense to get everything anyway for large tables. We should use a `paginator` here.

Reply:
Just checked unfortunately it seems `get_records` is not supporting paginate ...
<img width="1029" alt="Screen Shot 2022-08-26 at 9 27 57 AM" src="https://user-images.githubusercontent.com/107569566/186950561-4bcacff0-1ae1-4a8c-8111-3c7ef677cf80.png">

```        print(">>> can_paginate('get_records')")
        print(self.config.dynamodbstreams_client.can_paginate('get_records'))```

Reply:
Hmm.. bummer! We need to figure another way to get coverage I suppose.

Reply:
Actually, since this is a shard iterator, it has a built in mechanism for iteration. Essentially you get back a next shard iterator in the response and it will chain on until it exhausts that shard lineage.